### PR TITLE
CUR - Add comparison and some styling to curriculum screenshot script

### PR DESCRIPTION
**Pull request by BTitterington**
## Description

Music year: {owa_music_year}

- Added a visual comparison area to highlight and differences between the 2 images
- Added a menu to navigate quickly to screenshots
- Updated the styling to allow above
- I also was seeing visual differences where the images were stitched together during the screenshot. I updated the code so that the stitching part wasn't needed, which fixed the issue.
- There is a line to remove the cookie banner
- It now uses incognito mode as I thought that would be more reliable 

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: scripts/dev/curriculum/_commands/screenshot.ts [hunk 1]
```diff
@@ -1,109 +1,32 @@
...
-        x: 0,
-        y: i * viewportHeight,
-        width: viewportWidth,
-        height: Math.min(viewportHeight, totalHeight - i * viewportHeight),
-      },
-    });
-  }
+  await page.screenshot({
+    path: path,
+    captureBeyondViewport: true,
```
Comment (orangemug) [inline scripts/dev/curriculum/_commands/screenshot.ts:23]: Hmmm I was getting non-full pages captures with this option. Will verify on my machine shortly.

File: scripts/dev/curriculum/_commands/screenshot.ts [hunk 2]
```diff
@@ -1,109 +1,32 @@
...
-
 const screenshotPage = async (
   page: Page,
   url: string,
   path: string,
   logOpts: { id: string },
 ) => {
   await page.goto(url, {
-    waitUntil: "domcontentloaded",
+    waitUntil: "networkidle0",
```
[the remark this answers is not in the document]
Comment (BTitterington) [reply scripts/dev/curriculum/_commands/screenshot.ts:14]: The cookie consent banner was not getting removed from the screenshots until I changed this.

File: scripts/dev/curriculum/_commands/compare.ts [hunk 3]
```diff
@@ -38,60 +41,123 @@ export default async function compare(
...
+                  position: relative;
+                  overflow: hidden;
+                }
+                .diff-image{
+                  position: absolute;
+                  top: 0;
+                  left: 0;
+                  bottom: 0;
+                  right: 0;
+                  mix-blend-mode: difference;
```
Comment (orangemug) [inline scripts/dev/curriculum/_commands/compare.ts:95]: Oh that's smart 👏

File: scripts/dev/curriculum/_commands/screenshot.ts (hunk 1 above)
Comment (orangemug) [reply scripts/dev/curriculum/_commands/screenshot.ts:23]: Yup this works... no idea why it wasn't working previously 🤷

**Comment by orangemug** [comment]
Added a few small amendments in https://github.com/oaknational/Oak-Web-Application/pull/2854/commits/136dbfbfd0a046487a42f9861e1e55d490e93f35 else it works great. Thanks @BTitterington

**Review by orangemug — APPROVED** [review]

**Pull request merged**